src/modules/media/dto/media.dto.ts

```diff
@@ -1,224 +1,63 @@
-		},
-	})
-	duration?: RangeQueryDuration;
-
-	@Type(() => RangeQueryDate)
-	@IsOptional()
-	@ValidateNested()
-	@ApiPropertyOptional({
-		type: RangeQueryDate,
-		description: 'Specify upper and/or lower bounds for the creation date of the media item',
-		example: {
-			gte: '2020-09-01',
-			lte: '2020-12-31',
-		},
+		description: `The single value for the filter. If both multiValue and value are set, value is ignored.`,
 	})
-	created?: RangeQueryDate;
-
-	@Type(() => RangeQueryDate)
-	@IsOptional()
-	@ValidateNested()
-	@ApiPropertyOptional({
-		type: RangeQueryDate,
-		description: 'Specify upper and/or lower bounds for the publish date of the media item',
-		example: {
-			gte: '2020-09-01',
-			lte: '2020-12-31',
-		},
-	})
-	published?: RangeQueryDate;
+	value?: string;
 
 	@IsString()
-	@IsOptional()
-	@ApiPropertyOptional({
-		type: String,
-		description:
-			'Filter the results on the creator of the media item - TODO currently this requires an exact match',
-		required: false,
-	})
-	// TODO update filter for creator? -- currently requires an exact match (incl. case sensitivity)
-	creator?: string;
-
-	@IsArray()
-	@IsOptional()
-	@Transform((genre) => {
-		if (!isArray(genre.value)) {
-			throw new InternalServerErrorException('Genre should be an array');
-		}
-		return genre.value.map((g) => g.trim().toLowerCase());
-	})
-	@ApiPropertyOptional({
-		type: String,
-		description:
-			'Filter the results on the genre of the media item - TODO currently this requires an exact match',
-		required: false,
-		example: ['magazine'],
-	})
-	// TODO update filter for genre? -- currently requires an exact match and input should be lowercased
-	genre?: Array<string>;
-
-	@IsArray()
-	@IsOptional()
-	@Transform((keyword) => {
-		if (!isArray(keyword.value)) {
-			throw new InternalServerErrorException('Keywords should be an array');
-		}
-		return keyword.value.map((kw) => kw.trim().toLowerCase());
-	})
-	@ApiPropertyOptional({
+	@IsEnum(Operator)
+	@ApiProperty({
 		type: String,
-		description:
-			'Filter the results on the keywords of the media item - TODO currently this requires an exact match',
-		required: false,
-		example: ['belgium'],
+		description: `The query operator. Options are: ${Object.values(Operator).join(', ')}`,
 	})
-	// TODO case sensitive, works only if lowercased
-	keyword?: Array<string>;
-
-	@Type(() => AdvancedQuery)
-	@IsOptional()
-	@ValidateNested()
-	@ApiPropertyOptional({
-		type: AdvancedQuery,
-		description:
-			'An advanced filter on media item name (title). Filter on items contain or equal the given string (or opposite)',
-		required: false,
-		example: { contains: 'tv', containsNot: 'radio' },
-	})
-	name?: AdvancedQuery;
+	operator: Operator;
 }
 
 export class MediaQueryDto {
-	@Type(() => SearchFilters)
-	@IsOptional()
+	@Type(() => SearchFilter)
+	@IsArray()
 	@ValidateNested()
 	@ApiPropertyOptional({
-		type: SearchFilters,
-		description: 'Filters to query the media items',
+		type: SearchFilter,
```

Comment:
```suggestion
		type: () => [SearchFilter],
```

So swagger shows it as an array of objects:
![image](https://user-images.githubusercontent.com/1710840/158547544-cf115825-086a-494e-a37a-6ec28db994ed.png)